src/main/java/it/gov/pagopa/rtd/ms/rtdmsingestor/model/Transaction.java

```diff
@@ -22,58 +24,88 @@
 public class Transaction {
 
   @CsvBindByPosition(position = 0)
+  @Pattern(regexp = "[a-zA-Z0-9]{5}", message = "ABI length must be 5 alphanumeric char")
   String acquirerCode;
 
   @CsvBindByPosition(position = 1)
+  @Pattern(regexp = "[0-9]{2}", message = "Operation type length must match [0-9]{2}")
   String operationType;
 
   @CsvBindByPosition(position = 2)
+  @Pattern(regexp = "[0-9]{2}", message = "Circuit type length must match [0-9]{2}")
   String circuitType;
 
   @CsvBindByPosition(position = 3)
+  @Pattern(regexp = "[a-zA-Z0-9]{1,64}",
+      message = "HPAN length must be 64 alphanumeric alphanumeric char at max")
   String hpan;
 
+  //ISO8601
+  @NotNull
```

Comment:
Is there any downside in adding the @NotNull decorator on all non-nullable methods even if the @Pattern clearly prevent that? From a readability point-of-view could be an improvement, even if the implementation is already correct. What do you think?

Reply:
I deliberately removed every @NotNull annotation when a @Pattern annotation was present.
The ratio behind is that the validator, in presence of a null field, contained 2 violations (one for @NotNull and one for @Pattern) for the same field.
This behaviour polluted the log with a redundant reference to the same field.

I agree with you that specifying @NotNull is a clearer solution, but worsen the logging.
We could make a verbose violation evaluation step where we just collect a "distinct" of malformed fields, regardless of the source of the violation.

Reply:
In this case I think it's OK as it is, I was just wondering if there it was space for improving ;-)